src/session/session_api.c

```diff
@@ -2762,9 +2765,11 @@ __wt_open_internal_session(WT_CONNECTION_IMPL *conn, const char *name, bool open
 }
 
 #ifdef HAVE_UNITTEST
-int
-__ut_session_config_int(WT_SESSION_IMPL *session, const char *config)
-{
-    return (__session_config_int(session, config));
-}
+/* Disable temporarily as the test is not compatible with compiled configurations.
+ * int
+ * __ut_session_config_int(WT_SESSION_IMPL *session, const char *config)
+ * {
+ *    return (__session_config_int(session, config));
+ * }
```

Comment:
I spent a day trying to make the catch2 unit test work for compiled configuration, by attempting to convert traditional `const char *config` to `WT_CONF *conf` and pass it to `__session_config_int`. However, the existing session mock does not contain the right contents to support compiled configuration. 

I'm thinking to delay the catch2 unit test support work for now, and address it in a separate PR.

Reply:
[WT-15635](https://jira.mongodb.org/browse/WT-15635) was filed